lib/plug/conn.ex

```diff
@@ -460,12 +460,15 @@ defmodule Plug.Conn do
   otherwise `{:error, reason}`.
   """
   @spec chunk(t, body) :: {:ok, t} | {:error, term} | no_return
-  def chunk(%Conn{state: :chunked} = conn, ""), do: {:ok, conn}
   def chunk(%Conn{adapter: {adapter, payload}, state: :chunked} = conn, chunk) do
-    case adapter.chunk(payload, chunk) do
-      :ok                  -> {:ok, conn}
-      {:ok, body, payload} -> {:ok, %{conn | resp_body: body, adapter: {adapter, payload}}}
-      {:error, _} = error  -> error
+    case IO.iodata_length(chunk) do
```

Comment:
Which is better of course since it short circuits, but not really an issue when it runs so quickly anyways. 

My empty_iodata? looks like, (which may not be optmial since I eye balled it over lunch :P)
```
  def empty_iodata?(""), do: true
  def empty_iodata?([""]), do: true
  def empty_iodata?([]), do: true
  def empty_iodata?([head | tail]) do
    empty_iodata?(head) && empty_iodata?(tail)
  end
  def empty_iodata?(_), do: false